tests/acceptance/registries/discover-test.ts

```diff
@@ -0,0 +1,201 @@
+                order,
+                query: 'Foo',
+                filters: OrderedSet([
+                    new ShareTermsFilter('sources', 'OSF', 'OSF Registries'),
+                    new ShareTermsFilter('sources', 'ClinicalTrials.gov', 'ClinicalTrials.gov'),
+                ]),
+            },
+        }, {
+            name: 'Sort',
+            params: { sort: 'date_updated' },
+            expected: {
+                query: '',
+                order: new SearchOrder({
+                    ascending: true,
+                    display: 'registries.discover.order.modified_ascending',
+                    key: 'date_updated',
+                }),
+            },
+        }, {
+            name: 'Sort decending',
+            params: { sort: '-date_updated' },
+            expected: {
+                query: '',
+                order: new SearchOrder({
+                    ascending: false,
+                    display: 'registries.discover.order.modified_descending',
+                    key: 'date_updated',
+                }),
+            },
+        }, {
+            name: 'Sort validation',
+            params: { q: 'Not Empty', sort: '-date_moodified' },
+            expected: { order, query: 'Not Empty' },
+        }, {
+            name: 'Registration Types without OSF',
+            params: { q: 'What', page: 10, type: 'Foo|BAR' },
+            expected: { order, query: 'What', page: 10 },
+        }, {
+            // NOTE: Not currently validated :(
+            name: 'Registration Types',
+            params: { q: 'What', page: 10, provider: 'OSF', type: 'Foo|BAR' },
+            expected: {
+                order,
+                query: 'What',
+                page: 10,
+                filters: OrderedSet([
+                    new ShareTermsFilter('sources', 'OSF', 'OSF Registries'),
+                    new ShareTermsFilter('registration_type', 'Foo', 'Foo'),
+                    new ShareTermsFilter('registration_type', 'BAR', 'BAR'),
+                ]),
+            },
+        }, {
+            // NOTE: Not currently validated :(
+            name: 'Legacy Registration Types',
+            params: { q: 'What', page: 10, provider: 'OSF', type: 'FooORBAR' },
+            expected: {
+                order,
+                query: 'What',
+                page: 10,
+                filters: OrderedSet([
+                    new ShareTermsFilter('sources', 'OSF', 'OSF Registries'),
+                    new ShareTermsFilter('registration_type', 'Foo', 'Foo'),
+                    new ShareTermsFilter('registration_type', 'BAR', 'BAR'),
+                ]),
+            },
+        }];
+
+        assert.expect(2 + (testCases.length * 3));
+        const stub = sinon.stub(this.owner.lookup('service:share-search'), 'registrations').returns(results);
+
+        // Initial load so we don't have to deal with the aggregations loading
+        await visit('/registries/discover');
+        sinon.assert.calledTwice(stub);
```

Comment:
You can add a custom message:

```ts
assert.ok(stub.calledTwice, 'Called the share-search service's registrations method twice.')
```

(but I honestly haven't run the tests to see if the default message is adequate)